File: app/src/main/java/br/com/pedrosilva/tecnonutri/presentation/presenters/FeedItemPresenter.java
package br.com.pedrosilva.tecnonutri.presentation.presenters;

import br.com.pedrosilva.tecnonutri.domain.entities.FeedItem;
import br.com.pedrosilva.tecnonutri.presentation.presenters.base.BasePresenter;
import br.com.pedrosilva.tecnonutri.presentation.ui.BaseView;

/**
 * Created by psilva on 3/20/17.
 */

public interface FeedItemPresenter extends BasePresenter {
    interface View extends BaseView {
        void onLoadFeedItem(FeedItem feedItem);
        void onLoadFail(Throwable t);
        void onChangeLike(String feedHash, boolean like);
    }

    void reload();
    void changeLike(String feedHash, boolean liked);
    boolean isLiked(String feedHash);
}

File: app/src/main/kotlin/br/com/pedrosilva/tecnonutri/presentation/presenters/FeedItemPresenter.kt
package br.com.pedrosilva.tecnonutri.presentation.presenters

import com.pedrenrique.tecnonutri.domain.FeedItem
import br.com.pedrosilva.tecnonutri.presentation.presenters.base.BasePresenter
import br.com.pedrosilva.tecnonutri.presentation.ui.BaseView

interface FeedItemPresenter : BasePresenter {

    interface View : BaseView {

        fun onLoadFeedItem(item: FeedItem)

        fun onLoadFail(error: Throwable)

        fun onChangeLike(feedHash: String, like: Boolean)
    }

    fun reload()

    fun changeLike(feedHash: String, liked: Boolean)

    fun isLiked(feedHash: String): Boolean
}